File: app/src/main/java/com/gallery/cleaner/domain/model/MediaItem.kt
package com.gallery.cleaner.domain.model

import android.net.Uri

data class MediaItem(
    val id: Long,
    val uri: Uri,
    val name: String,
    val mimeType: String,
    val dateAdded: Long,
    val size: Long,
    val duration: Long? = null,
    val width: Int = 0,
    val height: Int = 0,
    val livePhotoInfo: LivePhotoInfo = LivePhotoInfo(),
    val isQueuedForDelete: Boolean = false
) {
    val isVideo: Boolean get() = mimeType.startsWith("video/")
    val isLivePhoto: Boolean get() = livePhotoInfo.type != LivePhotoType.NONE && !isVideo
}

File: ios/GalleryCleaner/Models/MediaItem.swift
import Foundation
import Photos

struct MediaItem: Identifiable, Hashable {
    let id: String          // PHAsset localIdentifier
    let asset: PHAsset
    let name: String
    let mimeType: String
    let dateAdded: Date
    let size: Int64
    let duration: TimeInterval?
    let width: Int
    let height: Int
    
    var isVideo: Bool { asset.mediaType == .video }
    var isLivePhoto: Bool { asset.mediaSubtypes.contains(.photoLive) }
    
    func hash(into hasher: inout Hasher) { hasher.combine(id) }
    static func == (lhs: MediaItem, rhs: MediaItem) -> Bool { lhs.id == rhs.id }
}

extension MediaItem {
    static func from(asset: PHAsset) -> MediaItem {
        let resources = PHAssetResource.assetResources(for: asset).first
        let fileName = resources?.originalFilename ?? "Unknown"
        let utType = resources?.uniformTypeIdentifier ?? ""
        let fileSize = resources?.value(forKey: "fileSize") as? Int64 ?? 0
        
        return MediaItem(
            id: asset.localIdentifier,
            asset: asset,
            name: fileName,
            mimeType: utType,
            dateAdded: asset.creationDate ?? Date(),
            size: fileSize,
            duration: asset.mediaType == .video ? asset.duration : nil,
            width: asset.pixelWidth,
            height: asset.pixelHeight
        )
    }
}
